Python/pywarpx/picmi.py

```diff
@@ -1809,14 +1805,13 @@ def initialize_inputs(self):
             # --- is the same on all processors.
             fields_to_plot = list(fields_to_plot)
             fields_to_plot.sort()
-            self.diagnostic.fields_to_plot = fields_to_plot
-
+            self.diagnostic.set_or_replace_attr('fields_to_plot', fields_to_plot)
         self.diagnostic.plot_raw_fields = self.plot_raw_fields
         self.diagnostic.plot_raw_fields_guards = self.plot_raw_fields_guards
         self.diagnostic.plot_finepatch = self.plot_finepatch
         self.diagnostic.plot_crsepatch = self.plot_crsepatch
-        self.diagnostic.write_species = self.write_species
-
+        if 'write_species' not in self.diagnostic.argvattrs:
+            self.diagnostic.write_species = False
```

Comment:
This looks good. Perhaps add a comment noting that the check is needed since ParticleDiagnostic will set write_species.